crates/bevy_math/src/direction.rs

```diff
@@ -446,6 +456,51 @@ impl Dir3 {
         let quat = Quat::IDENTITY.slerp(Quat::from_rotation_arc(self.0, rhs.0), s);
         Dir3(quat.mul_vec3(self.0))
     }
+
+    /// Returns `self` after an approximate normalization, assuming the value is already nearly normalized.
+    /// Useful for preventing numerical error accumulation.
+    ///
+    /// # Example
+    /// The following seemingly benign code would start accumulating errors over time,
+    /// leading to `dir` eventually not being normalized anymore.
+    /// ```
+    /// # use bevy_math::prelude::*;
+    /// fn system(mut dir: &mut Dir3) {
+    ///     let quaternion = Quat::from_euler(EulerRot::XYZ, 1.0, 2.0, 3.0);
+    ///     *dir = quaternion * *dir;
+    /// }
```

Comment:
Maybe this is nit-picky, but I don't love having systems or other Bevy terminology in `bevy_math`. This also isn't even a system, since `&mut Dir3` is not a valid system parameter.

Imo these math types shouldn't really have any knowledge that Bevy's ECS exists, and should be treated as standalone math types similar to Glam types (especially given we might upstream this method for `Quat`).

Maybe we could make this just a normal function and have a comment saying it's called repeatedly or something? Or just put the rotation in a loop

Reply:
How do you like the changed comment?